lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -1157,7 +1157,7 @@ void FMemModuleOp::getAsmBlockArgumentNames(
 
 /// Lookup the module or extmodule for the symbol.  This returns null on
 /// invalid IR.
-FModuleLike InstanceOp::getReferencedModule() {
+Operation *InstanceOp::getReferencedModule() {
   auto circuit = (*this)->getParentOfType<CircuitOp>();
   if (!circuit)
     return nullptr;
```

Comment:
I'm wondering if we shouldn't also provide an operation that returns a `FModuleLike` to encode more strongly that FIRRTL instances can only really instantiate `FModuleLike` ops.

Reply:
In the spirit of opaque types, I would keep this as an operation to permit FIRRTL to be composed with other dialects.